cdap-common/src/main/java/co/cask/cdap/common/conf/CConfigurationUtil.java

```diff
@@ -33,4 +35,18 @@ public static void copyTxProperties(CConfiguration cConf, org.apache.hadoop.conf
       }
     }
   }
+
+  public static void checkCConfValidity(CConfiguration cConf) {
+    // Checks to ensure that certain keys (e.g. "root.prefix") are valid as expected by CDAP.
+    checkAlphaNumberic(cConf, Constants.CFG_ROOT_NAMESPACE);
+    checkAlphaNumberic(cConf, Constants.Dataset.TABLE_PREFIX);
+  }
+
+  private static void checkAlphaNumberic(CConfiguration cConf, String key) {
+    String value = cConf.get(key);
+    Preconditions.checkNotNull(value, String.format("Entry of CConf with key: %s is null", key));
```

Comment:
Don't use `String.format` for Preconditions. Preconditions supports formatting already. Read the javadoc of Precondtions.

```
public static <T> T checkNotNull(T reference,
                                 @Nullable
                                 String errorMessageTemplate,
                                 @Nullable
                                 Object... errorMessageArgs)
```